src/box/tuple.h

```diff
@@ -359,6 +365,27 @@ struct PACKED tuple
 
 static_assert(sizeof(struct tuple) == 10, "Just to be sure");
 
+/** Set flag of the tuple. */
+static inline void
+tuple_set_flag(struct tuple *tuple, uint8_t flag)
```

Comment:
Let's make this code bullet-proof:

1. Add `enum tuple_flag` and use it here instead of `uint8_t` so that we can change the type without modifying this code:
   ```C
   static inline void
   tuple_set_flag(struct tuple *tuple, enum tuple_flag flag)
   ```
2. Store bit numbers instead of masks in `enum tuple_flag` so that one can't pass multiple flags to `tuple_set_flag` (this would be confusing):
   ```C
   enum tuple_flag {
       TUPLE_HAS_UPLOADED_REFS,
       TUPLE_IS_DIRTY,
       tuple_flag_MAX,
   };
   ```
3. Add a static assertion that ensures that all flags fit in `tuple::flags`:
   ```C
   static_assert(DIV_ROUND_UP(tuple_flag_MAX, 8) <= sizeof(((struct tuple *)0)->flags));
   ```